libbeat/autodiscover/providers/kubernetes/pod.go

```diff
@@ -63,7 +63,11 @@ func NewPodEventer(uuid uuid.UUID, cfg *common.Config, client k8s.Interface, pub
 	// Ensure that node is set correctly whenever the scope is set to "node". Make sure that node is empty
 	// when cluster scope is enforced.
 	if config.Scope == "node" {
-		config.Node = kubernetes.DiscoverKubernetesNode(logger, config.Node, kubernetes.IsInCluster(config.KubeConfig), client)
+		nd := &kubernetes.DiscoverKubernetesNodeOpts{ConfigHost: config.Node, Client: client, IsInCluster: kubernetes.IsInCluster(config.KubeConfig), HostUtils: &kubernetes.Hostdiscoveryutils{}}
```

Comment:
Nit. Consider splitting this in multiple lines so it is easier to read.
```suggestion
		nd := &kubernetes.DiscoverKubernetesNodeOpts{
		    ConfigHost: config.Node,
		    Client: client,
		    IsInCluster: kubernetes.IsInCluster(config.KubeConfig),
		    HostUtils: &kubernetes.Hostdiscoveryutils{},
		}
```